bindings/node.js/lib/kzg.js

```diff
@@ -6,6 +6,28 @@ const fs = require("fs");
 const path = require("path");
 const bindings = require("bindings")("kzg");
 
+/**
+ * Looks in the default locations for the trusted setup file.  This is for cases
+ * where the library is loaded without passing a trusted setup.  Should only be
+ * used for cases where the Ethereum official mainnet kzg setup is acceptable.
+ *
+ * @returns {string | undefined} - Filepath for trusted_setup.txt if found
+ */
+function getDefaultTrustedSetupFilepath() {
+  const locationsToSearch = [
+    // check the production bundle case first (this file in lib)
+    path.resolve(__dirname, "..", "deps", "c-kzg", "trusted_setup.txt"),
+    // check the development in-repo case second (this file in bindings/node.js/lib)
+    path.resolve(__dirname, "..", "..", "..", "src", "trusted_setup.txt"),
```

Comment:
Do we need this auto-detection logic? Why can't the user of the library always specify where the trusted setup is? Just worrying that the amount of `..` here makes this brittle.

What use cases are there, where someone loads a library and they don't know where the trusted setup file is, but the library knows?

Reply:
> Do we need this auto-detection logic? Why can't the user of the library always specify where the trusted setup is? Just worrying that the amount of .. here makes this brittle.

Sadly yes.  This an unfortunate pattern that shows up often in JS repos.  Generally src and artifacts are found in different places and usually prod code runs with artifacts (using transpiled JS) and unit test use src file running with pure TypeScript code and  `ts-node` under the hood somewhere.

> What use cases are there, where someone loads a library and they don't know where the trusted setup file is, but the library knows?

If a consumer/user supplies a value and it is invalid the library will throw and will not load a default config.  This way the consumer will not end up inadvertently loading a setup they were not expecting to load.  The default versions are ONLY used if no location is passed to the `loadTrustedSetup` function.

Reply:
> @asn-d6 sadly yes. This an unfortunate pattern that shows up often in JS repos. Generally src and artifacts are found in different places and usually prod code runs with artifacts (using transpiled JS) and unit test use src file running with pure TypeScript code and ts-node under the hood somewhere.

Hm. It's still not clear to me why we can't have prod code and unit tests always pass the right file path to the `loadTrustedSetup` function. This way, we could remove this `getDefaultTrustedSetupFilepath()` function and simplify the code.

Reply:
I will see what I can do to simplify things.  This is how I've learned to do an how I've always done it.  Will have to double check that I catch all the edge cases to make sure it works everywhere.  Stay-tuned @asn-d6.  Will try to get this pushed tomorrow morning (is almost midnight here).

Reply:
Will do!!  Sorry @asn-d6 will get this up asap

Reply:
Updated in https://github.com/ethereum/c-kzg-4844/pull/424